test/reasoning.spec.js

```diff
@@ -0,0 +1,99 @@
+/* eslint-env jest */
+
+const fs = require('fs');
+const { db } = require('../lib');
+const {
+  seedDatabase,
+  dropDatabase,
+  generateDatabaseName,
+  ConnectionFactory,
+} = require('./setup-database');
+
+const { reasoning, transaction } = db;
+
+describe('reasoning commands', () => {
+  const database = generateDatabaseName();
+  const conn = ConnectionFactory();
+
+  const beginTx = transaction.begin.bind(null, conn, database);
+  const commitTx = transaction.commit.bind(null, conn, database);
+
+  beforeAll(seedDatabase(database));
+  afterAll(dropDatabase(database));
+
+  it('should be able to check consistency', () => {
+    return reasoning.consistency(conn, database, {}).then(res => {
+      expect(res.status).toBe(200);
+      expect(res.body).toEqual('true');
+    });
+  });
+
+  it('should explain inferences', () => {
+    return reasoning
+      .explainInference(conn, database, '<urn:A> a <urn:B> .', {
+        contentType: 'text/turtle',
+      })
+      .then(res => {
+        expect(res.status).toBe(200);
+        expect(res.body.proofs).toBeTruthy();
+      });
+  });
+
+  it('should explain inconsistency', () => {
+    return reasoning.explainInconsistency(conn, database, {}).then(res => {
+      expect(res.status).toBe(200);
+      expect(res.body.proofs).toBeTruthy();
+    });
+  });
+
+  it('should explain inferences in a tx', () => {
+    return beginTx()
+      .then(res => {
+        expect(res.status).toBe(200);
+        expect(res.body).toBeGUID();
+        return reasoning.explainInferenceInTx(
+          conn,
+          database,
+          res.transactionId,
+          '<urn:A> a <urn:B> .',
+          { contentType: 'text/turtle' }
+        );
+      })
+      .then(res => {
+        expect(res.status).toBe(200);
+        expect(res.body.proofs).toBeTruthy();
+        // Makeshift 'finally'
+        return res;
+      })
+      .catch(res => res)
```

Comment:
A bit of a hack I found at https://stackoverflow.com/questions/35999072/what-is-the-equivalent-of-bluebird-promise-finally-in-native-es6-promises to hold over until ES2018 when native Promise gets a `finally` equivalent. Opening a tx without closing it is no bueno, so having the `then` after a `catch` ensures that it will be called no matter what.